Zend/zend_string.c

```diff
@@ -493,3 +497,45 @@ ZEND_API zend_string *zend_string_concat3(
 
 	return res;
 }
+
+// Copyright (c) 2008-2009 Bjoern Hoehrmann <bjoern@hoehrmann.de>
+// See http://bjoern.hoehrmann.de/utf-8/decoder/dfa/ for details.
+// https://stackoverflow.com/a/22135005/1320374
```

Comment:
This is under BSD licence so it should be pointed here as well or ideally new header created so it doesn't mix licenses in the single file. I have been actually using this in jsond for some time and I separated the headers for that.